how-to/containerd-kata.md

```diff
@@ -0,0 +1,357 @@
+```
+
+Or for the earlier versions of Kata Containers and containerd that don't support Runtime V2 (Shim API), you could use the alternative configuration:
+
+```toml
+    [plugins.cri.containerd]
+    [plugins.cri.containerd.default_runtime]
+      runtime_type = "io.containerd.runtime.v1.linux"
+      runtime_engine = "/usr/bin/kata-runtime"
+```
+
+### Configuration for cri-tools
+
+> **Note:** If you have skipped the [Install cri-tools](#install-cri-tools) section, you should probably skip this section too.
+
+Firstly, add the CNI configuration in the containerd configuration. 
+
+Here is the configuration if you installed CNI as the *[Install CNI plugins](#install-cni-plugins)* section.
+
+Put the CNI configuration as `/etc/cni/net.d/10-mynet.conf`:
+
+```json
+{
+	"cniVersion": "0.2.0",
+	"name": "mynet",
+	"type": "bridge",
+	"bridge": "cni0",
+	"isGateway": true,
+	"ipMasq": true,
+	"ipam": {
+		"type": "host-local",
+		"subnet": "172.19.0.0/24",
+		"routes": [
+			{ "dst": "0.0.0.0/0" }
+		]
+	}
+}
+```
+
+And reference the configuration directory through containerd `config.toml`:
+
+```toml
+[plugins.cri.cni]
+    # conf_dir is the directory in which the admin places a CNI conf.
+    conf_dir = "/etc/cni/net.d"
+```
+
+The configuration file of `crictl` command line tool in `cri-tools` locates at `/etc/crictl.yaml`:
+
+```yaml
+runtime-endpoint: unix:///var/run/containerd/containerd.sock
+image-endpoint: unix:///var/run/containerd/containerd.sock
+timeout: 10
+debug: true
+```
+
+## Run
+
+### Launch containers with ctr command line
+
+To run a container with Kata Containers through containerd command line, you could simply run:
+
+```bash
+$ sudo ctr run --runtime io.containerd.run.kata.v2 -t --rm docker.io/library/busybox:latest hello sh
+```
+
+which will launch a busybox container named as `hello` and it will be removed by `--rm` after it quits.
+
+### Launch Pods with crictl command line
+
+With `crictl` command line of `cri-tools`, you may specify runtime class with `-r` or `--runtime` flag,
+here is to launch Pod with `kata` runtime class with the pod in [the example](https://github.com/kubernetes-sigs/cri-tools/tree/master/docs/examples) 
+of `cri-tools`:
+
+```bash
+$ sudo crictl runp -r kata podsandbox-config.yaml
+36e23521e8f89fabd9044924c9aeb34890c60e85e1748e8daca7e2e673f8653e
+```
+
+And you may add container to the launched Pod:
+
+```bash
+$ sudo crictl create 36e23521e8f89 container-config.yaml podsandbox-config.yaml
+1aab7585530e62c446734f12f6899f095ce53422dafcf5a80055ba11b95f2da7
+```
+
+Then start it:
+
+```bash
+$ sudo crictl start 1aab7585530e6
+1aab7585530e6
+```
+
+In Kubernetes, you need create RuntimeClass resource and add RuntimeClass field in the Pod Spec (see the [document](https://kubernetes.io/docs/concepts/containers/runtime-class/)).
+
+And if RuntimeClass is not supported, you may have to use the following annotation in a Kubernetes pod to identify as an untrusted workload:
```

Comment:
Suggested rewrite: 
If RuntimeClass is not supported, you can use the following annotation in a Kubernetes pod to identify as an untrusted workload: